### ISSUE TITLE
Since only a few tilesets go into main game, mods should have the ability to add tiles to these.

### ISSUE THREAD

**Issue by Asmageddon** [CleverRaven/Cataclysm-DDA#19932]
Something that came up as an issue for me when adding heavy duty stowboards and corner variants of the ordinary stowboard.

**Comment by Leland** [comment]
No conversation in some time; closing for now. If anyone wants this reopened please ping me or just create a new issue.

**Comment by Asmageddon** [comment]
Well.. it's certainly not crucial, but I think that it's semi-important at least for mods to be able to add tiles, so their content isn't ASCII among graphical tiles.

**Comment by Asmageddon** [comment]
Hm, @Leland, I have nothing new to add, but finding this issue again by the way of things, I still think it's something that would be nice to have in the game.

**Comment by Leland** [comment]
@Asmageddon sure, we can keep this open for a while longer.

It's not a knock against the issue to close it – you typically have quality issues, and I agree with you on this one – it's just a simple fact that with well over 600 issues, Suggestions stay open only so long as people are talking about or working on them.

**Comment by lispcoc** [comment]
I agree, but I have no ideas that what kind of json structure is best written by mods. I seek your ideas to resolve this issue.

**Comment by mlangsdorf** [comment]
There was an earlier proposal for a "looks-like" or similar field, so you could add a new item and have tilesets re-use an existing image if one doesn't exist. It seemed conceptually simple to me and possibly relatively easy to program, but I haven't dug into it to be sure.

**Comment by mlangsdorf** [comment]
The problem with letting mods to add tile pngs is that the tileset loader doesn't know about mods and so it's going to be tricky to teach it to open tile.pngs and multiple tile_configs.json. The function is pretty well encapsulated, and I think it can already handle multiple tile.pngs, so that might not be too hard.

**Comment by lispcoc** [comment]
@mlangsdorf It's good solution! but I prefer "looks-like" field to be prioritize over default tile setting.

**Comment by mlangsdorf** [comment]
Do you mean the looks-like should override the assigned value from the tileset if the tileset has one? That's going to be a tough sell.

I thought the use case was:
mod_content = {
  "item": "hddoor",
  "looks_like": "door"
}
and then when you loaded Retrodays ISO that hasn't been updated in years, retrodays ISO makes hddoors look like doors. The code above should be an outline towards doing that.

if you mean a use case like this:
mod_content = {
  "item": "door",
  "copy-from": "door",
  "looks_like": "chicken_walker",
}
and when you load Chesthole, all the vehicle doors look like chicken walkers.
That may or may not be an interesting thing to do, but it's not something I intend to work on. But you're welcome to try it.

**Comment by lispcoc** [comment]
@mlangsdorf Isee and please forget it. I will try it and consider again.

**Comment by mlangsdorf** [comment]
went ahead and did a rough prototype of the solution I proposed, and it was good enough that I could create a new item "test_scythe", give it "looks_like": "scythe", and have the chesthole tileset display it as a scythe.
I went to test the recursive search through copy-from functions and then expand the logic to cover a few more types of things, but the basic idea is sound.

**Comment by mlangsdorf** [comment]
![success](https://user-images.githubusercontent.com/8007405/43897779-d046b98e-9ba2-11e8-9582-6bd0e95050d0.png)
Screen shot of test_scythe with looks_like scythe.

**Comment by Asmageddon** [comment]
Hm, not fully convinced if that invalidates this issue, but... well, it's better than nothing for sure.
